index_impl.go

```diff
@@ -228,7 +229,7 @@ func openIndexUsing(path string, runtimeConfig map[string]interface{}) (rv *inde
 		return nil, err
 	}
 
-	var im IndexMapping
+	var im *mapping.IndexMappingImpl
```

Comment:
wondering here offhand if the mapping unmarshaling might or is able to be moved fully into the mapping package, so that way mapping.IndexMappingImpl doesn't necessarily have to be exposed?